32bit build with GMP fails

```
$ make -s distclean ; ./configure && make -s clean && make -s 
[...]
Target CPU .................................. i686 AVX2, 32-bit LE
AES-NI support .............................. run-time detection
Target OS ................................... linux-gnu
Cross compiling ............................. no
Legacy arch header .......................... x86-sse.h

Optional libraries/features found:
Experimental code ........................... no
OpenMPI support (default disabled) .......... no
Fork support ................................ yes
OpenMP support .............................. yes
OpenCL support .............................. no
CUDA support ................................ no
Generic crypt(3) format ..................... yes
Rexgen (extra cracking mode) ................ no
GMP (PRINCE mode and faster SRP formats) .... yes
128-bit integer (faster PRINCE mode) ........ no
PCAP (vncpcap2john and SIPdump) ............. yes
BZ2 (gpg2john extra decompression logic) .... yes
Memory map (share/page large files) ......... yes

Development options (these may hurt performance when enabled):
Memdbg memory debugging settings ............ disabled
AddressSanitizer ("ASan") ................... disabled

Install missing libraries to get any needed features that were omitted.

Configure finished.  Now 'make clean && make -s' to compile.
/usr/bin/ar: creating aes.a
pp.c: In function ‘get_progress’:
pp.c:885:17: error: invalid operands to binary == (have ‘double’ and ‘struct __mpf_struct *’)
   if ((double)0 == count)
                 ^
pp.c:886:10: error: incompatible types when assigning to type ‘mpf_t’ from type ‘int’
     perc = 0;
          ^
make[1]: *** [pp.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [default] Error 2
```

--- a/src/mpz_int128.h
+++ b/src/mpz_int128.h
@@ -193,6 +193,7 @@ extern double __ntod(__uint128_t);
 #else
 #define mpf_set_z(x, y) x = y
 #endif // __MIC__
+#define mpf_sgn(F) ((F) < (double)0.0 ? -1 : (F) > (double)0.0)
 #define mpf_div(q, n, d) q = n / d
 #define mpf_clear(x) x = 0
 #define mpf_get_d(x) x

--- a/src/pp.c
+++ b/src/pp.c
@@ -882,9 +882,7 @@ static double get_progress(void)
   mpf_init(fpos); mpf_init(perc);
 
   mpf_set_z(fpos, rec_pos);
-  if ((double)0 == count)
-    perc = 0;
-  else
+  if (0 != mpf_sgn(count))
     mpf_div(perc, fpos, count);
   progress = 100.0 * mpf_get_d(perc);
 

